test/with_tmp_db

```diff
@@ -1,67 +1,114 @@
-    usage
+  echo "Please provide a command to be run with the temporary database."
+  usage
 fi
 
+rootdir="$(git rev-parse --show-toplevel)"
+fixturesdir="$rootdir/test/fixtures"
+
 # All data will be stored in a temporary directory.
 tmpdir="$(mktemp -d)"
 
-trap "rm -rf tmpdir" exit
+if [ -n "$keeptmp" ]; then
+  echo "The temporary directory at $tmpdir will be preserved after exit."
+else
+  trap "rm -rf tmpdir" sigint sigterm exit
+fi
 
-export PGDATA="$tmpdir"
-export PGHOST="$tmpdir"
-export PGUSER=postgrest_test_authenticator
-export PGDATABASE=postgres
+dblog="$tmpdir/db.log"
+setuplog="$tmpdir/setup.log"
 
-# Initialize a database cluster. We try to make it as independent as possible
-# from the host by specifying the timezone, locale and encoding.
-TZ=UTC initdb --no-locale --encoding=UTF8 --nosync -U "$PGUSER"
+log() {
+  echo "$1" >> "$setuplog"
+}
+
+mkdir -p "$tmpdir"/{db,socket}
 
-# Start the database cluster. Instead of listening on a local port, we will
-# listen on a unix domain socket. We will reuse the $PGDATA directory for that
-# socket.
-pg_ctl start -o "-F -c listen_addresses=\"\" -k $PGDATA"
+export PGDATA="$tmpdir/db"
+export PGHOST="$tmpdir/socket"
+export PGUSER=postgrest_test_authenticator
+export PGDATABASE=postgres
+export DB_URI="postgresql://$PGDATABASE?host=$PGHOST&user=$PGUSER"
+export POSTGREST_TEST_CONNECTION="$DB_URI"
+
+log "Initializing database cluster..."
+# We try to make the database cluster as independent as possible from the host
+# by specifying the timezone, locale and encoding.
+PGTZ=UTC initdb --no-locale --encoding=UTF8 --nosync -U "$PGUSER" --auth=trust \
+  >> "$setuplog"
+
+log "Starting the database cluster..."
+# Instead of listening on a local port, we will listen on a unix domain socket.
+pg_ctl -l "$dblog" start -o "-F -c listen_addresses=\"\" -k $PGHOST" \
+  >> "$setuplog"
+
+log "Waiting for the database cluster to be ready..."
+# Waiting is required for older versions of Postgres (< 10).
+until pg_isready >> "$setuplog"; do
+  sleep 0.1
+done
 
 stop() {
-    pg_ctl stop -m i
+  log "Stopping the database cluster..."
+  pg_ctl stop -m i >> "$setuplog"
+
+  if [ -n "$keeptmp" ]; then
+    echo "Keeping the temporary directory at $tmpdir"
+  else
     rm -rf "$tmpdir"
+  fi
 }
 
-trap stop exit
+trap stop sigint sigterm exit
 
-# Prepare the database for running the tests.
-psql <<EOF
-    create extension pgcrypto;
-    alter database $PGDATABASE set request.jwt.claim.id = '-1';# 
-    alter role $PGUSER set default_text_search_config to english;
-EOF
+log "Loading fixtures..."
+psql -v ON_ERROR_STOP=1 >> "$setuplog" << EOF
+  create extension pgcrypto;
+  alter database $PGDATABASE set request.jwt.claim.id = '-1';
+  alter role $PGUSER set default_text_search_config to english;
 
-export POSTGREST_TEST_CONNECTION="postgresql://$PGDATABASE?host=$PGDATA&user=$PGUSER"
+  \i $fixturesdir/database.sql
+  \i $fixturesdir/roles.sql
+  \i $fixturesdir/schema.sql
```

Comment:
Why the fixtures run in the script now? The haskell test helper already runs them:
https://github.com/PostgREST/postgrest/blob/master/test/SpecHelper.hs#L148-L155

Won't this be redundant?